src/DataStructures/VectorImpl.hpp

```diff
@@ -229,6 +229,33 @@ class VectorImpl
   }
   // @}
 
+  /*!
+   * \brief A common operation for checking the size and resizing a memory
+   * buffer if needed to ensure that it has the desired size. This operation is
+   * not permitted on a non-owning vector.
+   *
+   * \note This utility should NOT be used when it is anticipated that the
+   *   supplied buffer will typically be the wrong size (in that case, suggest
+   *   either manual checking or restructuring so that resizing is less common).
+   *   This uses `UNLIKELY` to perform the check most quickly when the buffer
+   *   needs no resizing, but will be slower when resizing is common.
+   */
+  void SPECTRE_ALWAYS_INLINE
+  unlikely_destructive_resize(const size_t new_size) noexcept {
```

Comment:
I think the `unlikely` part can be left out because it doesn't change the actual result of the function :)